### PR TITLE
(service-client): Add constructors in registry manager to accept aad …

### PR DESCRIPTION
**Pull request by vinagesh**
…and sas tokens.

## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

### CODE REVIEW

File: iothub/service/src/HttpRegistryManager.cs [hunk 1]
```diff
@@ -54,12 +54,12 @@ internal class HttpRegistryManager : RegistryManager
         private IHttpClientHelper _httpClientHelper;
         private readonly string _iotHubName;
 
-        internal HttpRegistryManager(IotHubConnectionString connectionString, HttpTransportSettings transportSettings)
+        internal HttpRegistryManager(IotHubConnectionProperties connectionProperties, HttpTransportSettings transportSettings)
```
Comment (vinagesh) [inline iothub/service/src/HttpRegistryManager.cs:57]: All these properties are now in a base class, making the HttpRegistryManager independent of the auth type.

**Review by timtay-microsoft — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/service/src/RegistryManager.cs [hunk 2]
```diff
@@ -46,6 +53,66 @@ public static RegistryManager CreateFromConnectionString(string connectionString
...
+            }
+
+            if (credential == null)
+            {
+                throw new ArgumentNullException($"{nameof(credential)} is null");
+            }
+
+            TlsVersions.Instance.SetLegacyAcceptableVersions();
+
+            IotHubTokenCrendentialProperties tokenCredentialProperties = new IotHubTokenCrendentialProperties(hostName, credential);
```
Comment (drwill-ms) [inline iothub/service/src/RegistryManager.cs:82]: var

File: iothub/service/src/RegistryManager.cs [hunk 3]
```diff
@@ -46,6 +53,66 @@ public static RegistryManager CreateFromConnectionString(string connectionString
...
+            }
+
+            if (credential == null)
+            {
+                throw new ArgumentNullException($"{nameof(credential)} is null");
+            }
+
+            TlsVersions.Instance.SetLegacyAcceptableVersions();
+
+            IotHubSasCredentialProperties sasCredentialProperties = new IotHubSasCredentialProperties(hostName, credential);
```
Comment (drwill-ms) [inline iothub/service/src/RegistryManager.cs:110]: var

File: iothub/service/src/RegistryManager.cs [hunk 4]
```diff
@@ -46,6 +53,66 @@ public static RegistryManager CreateFromConnectionString(string connectionString
...
+            {
+                throw new ArgumentNullException($"{nameof(hostName)} is null or empty");
+            }
+
+            if (credential == null)
+            {
+                throw new ArgumentNullException($"{nameof(credential)} is null");
+            }
+
+            TlsVersions.Instance.SetLegacyAcceptableVersions();
```
Comment (drwill-ms) [inline iothub/service/src/RegistryManager.cs:80]: Why call this? It does nothing when compiled with .NET 4.5.1, and this code block is in a !NET451.

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/src/RegistryManager.cs (hunk 4 above)
Comment (vinagesh) [reply iothub/service/src/RegistryManager.cs:80]: I had not seen the implementation so did not realize that. Will remove it.

**Pull request merged**